examples/common/python/utility/signature.py

```diff
@@ -315,4 +315,106 @@ def verify_signature(self, input_json, verification_key):
             return SignatureStatus.INVALID_SIGNATURE_FORMAT
 
 #---------------------------------------------------------------------------------------------
+    def verify_update_receipt_signature(self, input_json):
+        """
+        Function to verify the signature of work order receipt update
+        Parameters:
+            - input_json is dictionary contains payload returned by the
+              WorkOrderReceiptUpdateRetrieve API as define EEA spec 7.2.7
+        Returns enum type SignatureStatus
+        """
+        input_json_params = input_json['result']
+
+        concat_string = input_json_params["workOrderId"] + \
+            str(input_json_params["updateType"]) + \
+            input_json_params["updateData"]
+        concat_hash = bytes(concat_string, 'UTF-8')
+        final_hash = crypto.compute_message_hash(concat_hash)
+        signature = input_json_params["updateSignature"]
+        verification_key = input_json_params["receiptVerificationKey"]
+
+        try:
+            _verifying_key = crypto.SIG_PublicKey(verification_key)
+        except Exception as error:
+            logger.info("Error in verification key : %s", error)
+            return SignatureStatus.INVALID_VERIFICATION_KEY
+
+        decoded_signature = crypto.base64_to_byte_array(signature)
+        sig_result = _verifying_key.VerifySignature(final_hash, decoded_signature)
+
+        if sig_result == 1:
+            return SignatureStatus.PASSED
+        elif sig_result == 0:
+            return SignatureStatus.FAILED
+        else:
+            return SignatureStatus.INVALID_SIGNATURE_FORMAT
+
+#---------------------------------------------------------------------------------------------
+    def verify_create_receipt_signature(self, input_json):
+        """
+        Function to verify the signature of work order receipt create
+        Parameters:
+            - input_json is dictionary contains request payload of
+              WorkOrderReceiptRetrieve API as define EEA spec 7.2.2
+        Returns enum type SignatureStatus
+        """
+        input_json_params = input_json['params']
+
+        concat_string = input_json_params["workOrderId"] + \
+            input_json_params["workerServiceId"] + \
+            input_json_params["workerId"] + \
+            input_json_params["requesterId"] + \
+            str(input_json_params["receiptCreateStatus"]) + \
+            input_json_params["workOrderRequestHash"] + \
+            input_json_params["requesterGeneratedNonce"]
+        concat_hash = bytes(concat_string, "UTF-8")
+        final_hash = crypto.compute_message_hash(concat_hash)
+        signature = input_json_params["requesterSignature"]
+        verification_key = input_json_params["receiptVerificationKey"]
```

Comment:
both verify_receipt_create_signature and verify_receipt_update_signature look similar except calculating hash on different params. Can we create a utility function for this?

Reply:
No both are different.